src/Carbon/CarbonInterval.php

```diff
@@ -2752,4 +2763,14 @@ public function ceil($precision = 1)
     {
         return $this->round($precision, 'ceil');
     }
+
+    private function needsDeclension(string $mode, int $index, int $parts): bool
+    {
```

Comment:
or just `return $mode !== 'last' || $index === $parts - 1;` ?

Reply:
I think the switch will keep easier to add more modes and more clear about what is the grammar rule.